src/material-experimental/dialog/dialog-ref.ts

```diff
@@ -0,0 +1,137 @@
+    protected _overlayRef: OverlayRef,
+    protected _containerInstance: CdkDialogContainer,
+    readonly id: string = `dialog-${uniqueId++}`) {
+
+    // If the dialog has a backdrop, handle clicks from the backdrop.
+    if (_containerInstance._config.hasBackdrop) {
+      _overlayRef.backdropClick().subscribe(() => {
+        if (!this.disableClose) {
+          this.close();
+        }
+      });
+    }
+
+    this.beforeClose().subscribe(() => {
+      this._overlayRef.detachBackdrop();
+    });
+
+    this.afterClosed().subscribe(() => {
+      this._overlayRef.detach();
+      this._overlayRef.dispose();
+      this.componentInstance = null!;
+    });
+  }
+
+  /** Gets an observable that emits when the overlay's backdrop has been clicked. */
+  backdropClick(): Observable<void> {
+    return this._overlayRef.backdropClick();
+  }
+
+  /**
+   * Close the dialog.
+   * @param dialogResult Optional result to return to the dialog opener.
+   */
+  close(dialogResult?: any): void {
+    this._result = dialogResult;
+    this._containerInstance._startExiting();
+  }
+
+  /**
+   * Updates the dialog's position.
+   * @param position New dialog position.
+   */
+  updatePosition(position?: DialogPosition): this {
+    let strategy = this._getPositionStrategy();
+
+    if (position && (position.left || position.right)) {
+      position.left ? strategy.left(position.left) : strategy.right(position.right);
+    } else {
+      strategy.centerHorizontally();
+    }
+
+    if (position && (position.top || position.bottom)) {
+      position.top ? strategy.top(position.top) : strategy.bottom(position.bottom);
+    } else {
+      strategy.centerVertically();
+    }
+
+    this._overlayRef.updatePosition();
+
+    return this;
+  }
+
+  /**
+   * Gets an observable that emits when keydown events are targeted on the overlay.
+   */
+  keydownEvents(): Observable<KeyboardEvent> {
+    return this._overlayRef.keydownEvents();
+  }
+
+  /**
+   * Updates the dialog's width and height.
+   * @param width New width of the dialog.
+   * @param height New height of the dialog.
+   */
+  updateSize(width: string = 'auto', height: string = 'auto'): this {
+    this._getPositionStrategy().width(width).height(height);
+    this._overlayRef.updatePosition();
+    return this;
+  }
+
+  /** Fetches the position strategy object from the overlay ref. */
+  private _getPositionStrategy(): GlobalPositionStrategy {
+    return this._overlayRef.getConfig().positionStrategy as GlobalPositionStrategy;
+  }
+
+  /** Gets an observable that emits when dialog begins opening. */
+  beforeOpen(): Observable<void> {
+    return this._containerInstance._beforeEnter.asObservable();
+  }
+
+  /** Gets an observable that emits when dialog is finished opening. */
+  afterOpen(): Observable<void> {
+    return this._containerInstance._afterEnter.asObservable();
+  }
+
+  /** Gets an observable that emits when dialog begins closing. */
+  beforeClose(): Observable<any> {
```

Comment:
`any` -> generic type (based on @crisbeto's recent PR that adds a generic for this)